src/App.jsx

```diff
@@ -1,33 +1,64 @@
+import { useState } from 'react';
+import moviesFromServer from './api/movies.json';
+
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
-import moviesFromServer from './api/movies.json';
 
-export const App = () => (
-  <div className="page">
-    <div className="page-content">
-      <div className="box">
-        <div className="field">
-          {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
-          <label htmlFor="search-query" className="label">
-            Search movie
-          </label>
-
-          <div className="control">
-            <input
-              type="text"
-              id="search-query"
-              className="input"
-              placeholder="Type search word"
-            />
+const includesQuery = (text, query) => {
+  const preparedText = text.toLowerCase();
+  const preparedQuery = query.toLowerCase();
+
+  return preparedText.includes(preparedQuery);
+};
+
+function getPreparedMovies(movies, query) {
+  const queryToLowerCase = query.trim();
+
+  return movies.filter(({ title, description }) => {
+    const hasSuitableTitle = includesQuery(title, queryToLowerCase);
+    const hasSuitableDescription = includesQuery(description, queryToLowerCase);
+
+    return hasSuitableTitle || hasSuitableDescription;
+  });
+}
+
+export const App = () => {
+  const [query, setQuery] = useState('');
+  const visibleMovies = getPreparedMovies(
+    moviesFromServer,
+    query,
+  );
+
+  return (
+    <div className="page">
+      <div className="page-content">
+        <div className="box">
+          <div className="field">
+            {/* eslint-disable-next-line jsx-a11y/label-has-associated-control */}
+            <label htmlFor="search-query" className="label">
+              Search movie
+            </label>
+
+            <div className="control">
+              <input
+                type="text"
+                id="search-query"
+                className="input"
+                placeholder="Type search word"
+                onChange={(event) => {
+                  setQuery(event.currentTarget.value);
+                }}
```

Comment:
In `React` it is better to use `event.target`

[Difference between target and currentTarget](https://stackoverflow.com/questions/10086427/what-is-the-exact-difference-between-currenttarget-property-and-target-property)
```suggestion
                onChange={(event) => setQuery(event.target.value)}
```